### PR TITLE
refactor: use TemporaryDirectory where possible

### PR DESCRIPTION
**Pull request by mrossinek**
### Summary

Instead of using a NamedTemporaryFile and juggle its existence state,
this commit refactors all scenarios to use a TemporaryDirectory within
which the file names can be controlled more easily.
The reason for this refactoring was to guard against possible runtime
conflicts when multiple processes might be attempting to access the same
temporary file name, after the object has been unlinked (which might
free up the temporary name for reuse).

### Details and comments

Thanks @woodsp-ibm for this suggestion :+1:

### CODE REVIEW

File: test/test_logging.py [hunk 1]
```diff
@@ -110,22 +110,19 @@ def test_logging_to_default_handler(self):
     def test_logging_to_file(self):
```
Comment (mrossinek) [inline test/test_logging.py:110]: I don't quite understand why the changes to this method make the test which gets executed after this one (namely `test_logging_to_handler`) fail..

@manoelmarques could you please take a look at this? I don't seem to find the reason for this cross-test dependence

Comment (manoelmarques) [reply test/test_logging.py:110]: Fixed the unit test method

**Review by manoelmarques — APPROVED** [review]
